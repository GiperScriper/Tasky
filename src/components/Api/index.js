import db, { firebase } from '@/firebase';

const Api = {
  createUser: ({ email, password }) => {
    return firebase.auth().createUserWithEmailAndPassword(email, password);
  },
  saveUser: ({ uid, displayName: name }) => {
    db.ref('users')
      .child(uid)
      .set({ name });
  },
};

export default Api;
