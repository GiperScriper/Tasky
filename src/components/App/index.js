import React, { Component } from 'react';
import AppRouter from 'routers/AppRouter';

import './App.css';

class App extends Component {
  render() {
    return (
      <div>
        <AppRouter />
      </div>
    );
  }
}

export default App;
