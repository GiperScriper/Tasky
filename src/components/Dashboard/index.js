import React from 'react';

export const Dashboard = () => {
  console.log('render Dashboard');
  return (
    <div>
      <h1>Dashboard</h1>
    </div>
  );
};

export default Dashboard;
