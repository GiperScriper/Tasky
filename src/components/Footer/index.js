import React from 'react';

export default () => {
  console.log('render Footer');
  return (
    <footer>
      <h1>footer</h1>
    </footer>
  );
};
