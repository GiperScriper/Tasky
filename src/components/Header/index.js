import React from 'react';
import { NavLink } from 'react-router-dom';
import './header.scss';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEarlybirds } from '@fortawesome/free-brands-svg-icons';
import { faBell, faCog, faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
import userImage from '@/assets/images/user.png';

const Header = () => {
  console.log('render Header');
  return (
    <header className="header">
      <div className="header__logo">
        <FontAwesomeIcon icon={faEarlybirds} />
      </div>
      <ul className="header__menu">
        <li>
          <NavLink exact className="header__menu-link" activeClassName="header__menu-link--active" to="/">
            Dashboard
          </NavLink>
        </li>
        <li>
          <NavLink className="header__menu-link" activeClassName="header__menu-link--active" to="/projects">
            Projects
          </NavLink>
        </li>
      </ul>
      <div className="header__notice">
        <button className="button button--transparent header__notice-button">
          <FontAwesomeIcon icon={faBell} />
          <span className="header__notice-indicator" />
        </button>
        <button className="button button--transparent header__notice-button">
          <FontAwesomeIcon icon={faQuestionCircle} />
          <span className="header__notice-indicator" />
        </button>
        <button className="button button--transparent header__notice-button">
          <FontAwesomeIcon icon={faCog} />
        </button>
      </div>
      <div className="header__user">
        <img src={userImage} className="header__user-image" alt="User" />
      </div>
    </header>
  );
};

export default Header;
