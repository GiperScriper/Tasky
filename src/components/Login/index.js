import React from 'react';
import { Form, Field } from 'react-final-form';
import authService from 'helpers/authService';
import { connect } from 'react-redux';
import { login } from 'actions/auth';
import { renderInputWithError } from 'helpers/validation';

import './login.scss';

class loginForm extends React.Component {
  onSubmit = data => {
    authService.setToken(data.login);
    this.props.login('test data');
    this.props.history.push('/');
    // TO-DO signin request
  };

  render() {
    return (
      <div className="login__container">
        <div className="login__title">Tasky</div>
        <div className="login__bg">
          <Form
            onSubmit={this.onSubmit}
            render={({ handleSubmit, pristine, invalid }) => (
              <form name="loginForm" onSubmit={handleSubmit} className="login__form">
                <Field name="login" type="text" placeholder="Login" component={renderInputWithError} />
                <Field name="password" type="password" placeholder="Password" component={renderInputWithError} />
                <button type="submit" className="login__button">
                  Sign In
                </button>
              </form>
            )}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  login: data => dispatch(login(data)),
});

export default connect(
  undefined,
  mapDispatchToProps
)(loginForm);
