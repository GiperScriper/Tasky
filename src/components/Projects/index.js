import React from 'react';

const Projects = () => (
  <div>
    <h1>Projects</h1>
  </div>
);

export default Projects;
