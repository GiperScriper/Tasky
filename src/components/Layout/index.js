import React from 'react';
import { Route } from 'react-router-dom';

import Header from 'components/Header';
import Dashboard from 'components/Dashboard';
import Projects from 'components/Projects';

export default () => (
  <div>
    <Header />
    <Route path="/" component={Dashboard} exact />
    <Route path="/projects" component={Projects} exact />
  </div>
);
