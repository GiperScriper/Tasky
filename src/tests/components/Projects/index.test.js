import React from 'react';
import { shallow } from 'enzyme';
import { Projects } from '@/components/Projects';

describe('<Projects />', () => {
  it('should render Projects correctly', () => {
    const wrapper = shallow(<Projects />);
    expect(wrapper).toMatchSnapshot();
  });
});
