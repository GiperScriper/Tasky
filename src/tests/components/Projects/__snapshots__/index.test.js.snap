// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render Projects correctly 1`] = `
<React.Fragment>
  <h1>
    Projects
  </h1>
  <Component
    projects={
      Array [
        Object {
          "description": "TO-DO: create projects list",
          "id": 1,
          "title": "Tasky",
        },
        Object {
          "description": "some sample project",
          "id": 2,
          "title": "Sample",
        },
      ]
    }
  />
</React.Fragment>
`;
